### ISSUE TITLE
SC1008: Add `#!/bin/false` for Sourced scripts

### ISSUE THREAD

**Issue by anishmittal2020** [koalaman/shellcheck#1478]
From [Shebang (Unix) - Wikipedia](https://en.wikipedia.org/wiki/Shebang_(Unix)#Examples):

> `#!/bin/false` – Do nothing, but return a non-zero [exit status](https://en.wikipedia.org/wiki/Exit_status "Exit status"), indicating failure. Used to prevent stand-alone execution of a script file intended for execution in a specific context, such as by the '.' command from sh/bash, 'source' from csh/tcsh, or as a .profile, .cshrc, or .login file.

Please add `#!/bin/false`.

**Comment by koalaman** [comment]
This works when you specify the shell in some other way, but this warning was not very helpful, and on top of that it lacked a wiki page! I've updated the message in 2ea229315 to suggest using a directive, and I created [a wiki page](https://github.com/koalaman/shellcheck/wiki/SC1008) demonstrating how.

tl;dr: Just specify the shell you expect will source the file, and the warning will be omitted:

```
#!/bin/false
# shellcheck shell=bash
echo "Hello World"
```

Thanks for pointing this out!

**Comment by anishmittal2020** [comment]
@koalaman 

```bash
$ shellcheck -V
ShellCheck - shell script analysis tool
version: 0.4.7
```

I tried what you suggested. i.e I did as you suggested:

```
#!/bin/false
# shellcheck shell=bash
```
 But this gives this error

```bash
$ shellcheck ~/.bash_profile

In /home/nikhil/.bash_profile line 1:
#!/bin/false
^-- SC1008: This shebang was unrecognized. Note that ShellCheck only handles sh/bash/dash/ksh.
```

But  when I switch the first 2 lines as follows:

```bash
# shellcheck shell=bash
#!/bin/false
```

then I don't get any error. 

But I think that the order should be that the first line is shebang.

**Comment by koalaman** [comment]
Oh, you're right. The directive not working as the second line was issue #974 fixed for v0.6.0. 

Unfortunately there's no great workaround for previous versions, and it can't be silenced with a directive or `-s` because that was exactly the bug being fixed. You could consider upgrading, or maybe use `#!/bin/bash -n` which would be equivalent to `#!/bin/true`

**Comment by JuPlutonic** [comment]
Actually 
```bash
#!/bin/false
# shellcheck shell=bash
```
works

**Comment by brother** [comment]
@JuPlutonic indeed. you are replying to closed issue where there has been no comments for two years and the last comment was the author stating that old versions does not cope with this. 
Using shellcheck versions post 2018 solves most of this. When using older versions, be aware and consider upgrading. There has been other useful updates in the mean time.
